### PR TITLE
Updated Controller, Model, and DataStore

### PR DESCRIPTION
**Pull request by mhypolit**
Data store was mapping data for role and user separately, in order for front end to work correctly role and user data must be pulled together in data store.  Data Store still also retain separate data for role in order to just pull roles or one role. A userRole controller was created in order to pull just a users roles or one role.
Solves #4

### CODE REVIEW

File: Cedr-Project-iON/Controllers/Users/UserCtrl.cs [hunk 1]
```diff
@@ -13,38 +13,65 @@ public class UserCtrl : Controller
...
         public IActionResult AddUser([FromBody] UsersDTO users)
         {
-            Console.Write(users); // here for testing perameter
+
+            if (users == null)
+            {
+                return BadRequest();
+            }
 
             var user = IonDataStore.Current.Users;
```
Comment (hollywoodno) [inline Cedr-Project-iON/Controllers/Users/UserCtrl.cs]: Small bugs introduced due to the naming of your 'user' and 'users' variables. It looks like you mean to accept from the request a 'UsersDTO' as '_user_'. When you call your data store, you mean to get back a list of _'users'_. So swapping the variable names should clear up the issue.

Comment (hollywoodno) [reply Cedr-Project-iON/Controllers/Users/UserCtrl.cs]: As another note, if you define `users = IonDataStore.Current.Users,`  as class property, then you can access the list of users in all of your methods via 'users'.

File: Cedr-Project-iON/Controllers/Users/UserCtrl.cs [hunk 2]
```diff
@@ -13,38 +13,65 @@ public class UserCtrl : Controller
...
             };
 
-            user.Add(tempResult);
-            return Ok(tempResult);
+            user.Add(newUser);
+     
+            return Created("NewUser", newUser);
         }
 
         [HttpPatch("{userId}")]
```
Comment (hollywoodno) [inline Cedr-Project-iON/Controllers/Users/UserCtrl.cs]: Is it possible that var `userToPatch = new Users .... `and `userPatchDoc.ApplyTo(userToPatch, ModelState)` perform the same action? Can we just instead write `userPatchDoc.ApplyTo(userFromDataStore, ModelState)`?

Comment (mhypolit) [reply Cedr-Project-iON/Controllers/Users/UserCtrl.cs]: Refactored to what comment suggested

File: Cedr-Project-iON/Controllers/Users/UserCtrl.cs (hunk 1 above)
Comment (mhypolit) [reply Cedr-Project-iON/Controllers/Users/UserCtrl.cs]: Update user and users variables

**Pull request merged**